decidim-core/app/controllers/decidim/participatory_processes_controller.rb

```diff
@@ -6,15 +6,14 @@ module Decidim
   # A controller that holds the logic to show ParticipatoryProcesses in a
   # public layout.
   class ParticipatoryProcessesController < ApplicationController
-    include NeedsParticipatoryProcess
-
     layout "layouts/decidim/participatory_process", only: [:show]
 
-    skip_after_action :verify_participatory_process, only: [:index]
+    before_action -> { extend(NeedsParticipatoryProcess) }, only: [:show]
```

Comment:
It can. As a matter of fact, that's what was done previously. But it was prone to problems: see the comments in 792f250. By extending at the class level, we were "extending too much", causing the problems I described.

I think this approach makes sense since it allows extending at the instance level when including the module does not make sense for all actions (like here), but also allows to include the module in the traditional way when it does make sense (e.g., for resources nested under a participatory process).